### PR TITLE
merge branches

### PR DESCRIPTION
**Pull request by melineWb**

### CODE REVIEW

File: client/src/app.config.router.js [hunk 1]
```diff
@@ -20,9 +23,30 @@ module.exports = function(app) {
...
   app.component('teacherComponent', teacherComponent);
 
+  app.controller('homeController', homeController);
+
   app.config(['$stateProvider', '$urlRouterProvider', function($stateProvider, $urlRouterProvider) {
     $stateProvider
 
+        .state('main', {
+          url: '/',
+          template: require('./states/main/template.html'),
```
Comment (statyan) [inline client/src/app.config.router.js:33]: Why not components ?

Comment (dmk1111) [reply client/src/app.config.router.js:33]: this is copy from my Auth draft, will redo ASAP for next pull request

File: client/src/components/fedbackPopUp/controller.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+function fedbackPopUpController($http, $scope, $element) {
+  let self = this;
+  self.$element = $element;
+
+  $scope.$parent.openPopUpClick = function(id){
+    let idLecture = '#fb'+id;
```
Comment (statyan) [inline client/src/components/fedbackPopUp/controller.js]: Not good idea to set property to the parent from one of its children. Imaging what will be if you will have two popups declared in parent template. The second one will overwrite $parent.openPopupClick method.
The better way - to make separate service with method shoPopup(id), inject this service into parent controller and call this method.

File: client/src/components/index.js [hunk 3]
```diff
@@ -1,10 +1,46 @@
...
+  app.component('fedbackPopUp', fedbackPopUpComponent);
+
+  const recentTasksController = require("./tasks/recentTasksController.js");
+  app.component('recentTaskItem', {
+    bindings: {
+      task: '='
+    },
+    template: require('./tasks/template.html'),
+    controller: recentTasksController,
+    controllerAs: 'recentTaskItem'
```
Comment (statyan) [inline client/src/components/index.js]: we can aviod decalring controllerAs for each component. There will be $ctrl by default accessible in template

**Review by statyan — APPROVED** [review]

**Review by dmk1111 — APPROVED** [review]
approved

**Pull request merged**